salalib/linkutils.cpp

```diff
@@ -72,4 +72,29 @@ namespace depthmapX {
         }
         return mergePixelPairs;
     }
+
+    void mergePixelPairs(std::vector<PixelRefPair>& links, PointMap& currentMap) {
+
+        // check if any link pixel already exists on the map
+        for (size_t i = 0; i < links.size(); i++)
+        {
+            PixelRefPair link = links[i];
+            if(currentMap.isPixelMerged(link.a)
+                    || currentMap.isPixelMerged(link.b))
+            {
+                // one of the cells is already on the map
+                std::stringstream message;
+                message << "Link pixel found that is already linked on the map "
+                        << "(line " << i << ")"
+                        << flush;
+                throw depthmapX::RuntimeException(message.str().c_str());
```

Comment:
If it appears twice on the list it will be caught by getLinksFromMergeLines. But indeed I can't find very good reasoning to have this as a separate loop in mergePixelPairs, apart from the possibility of using it without getLinksFromMergeLines. In that case maybe it's better if I bring over all the checks into mergePixelPairs and rename the other function to "pixelateMergeLines" without any checking. What do you think?